src/resources/Api.ts

```diff
@@ -78,16 +98,17 @@ export class Api {
         Tags: this.getTagsConfig(),
       },
     };
-
-    merge(
-      endpointResource.Properties,
-      this.compileAuthenticationProvider(this.config.authentication),
-    );
+    if (this.config.authentication) {
```

Comment:
`authentication` is always required in this context. 
There is probably some TS magic to do (i.e. a type union type or something)